python/private/pypi/parse_requirements.bzl

```diff
@@ -207,22 +207,46 @@ def parse_requirements(
         # Return normalized names
         ret_requirements = ret.setdefault(normalize_name(whl_name), [])
 
+        all_platforms = []
+        common_sdist = None
+
         for r in sorted(reqs.values(), key = lambda r: r.requirement_line):
             whls, sdist = _add_dists(
                 requirement = r,
                 index_urls = index_urls.get(whl_name),
                 logger = logger,
             )
 
```

Comment:
The idea of adding a common sdist feels like a hack. Maybe having a different algorithm/grouping would be better. Maybe having per sha grouping could be better.

Reply:
Took a stab at changing it to use the sha instead.